module2.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+var myModule = (function () {
+// Private
+  var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday',
+  'Thursday', 'Friday', 'Saturday'];
+  var months = ['January', 'February', 'March', 'April', 'May',
+   'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+  var myDate = null;
+
+  function setDate (n) {
+    if (n === undefined) {
+      myDate = new Date();
+
+      return myDate;
+    } else if (n instanceof Date) {
+      myDate = n;
+
+      return myDate;
+    } else {
```

Comment:
The only other input you want to use to instantiate a new Date with would be a number. By defaulting to passing your input to the Date constructor, you can end up with invalid dates, for example, if the user passed in a string such as 'abcd'